modules/shared/src/main/scala/org/tessellation/cutoff/LogarithmicOrdinalCutoff.scala

```diff
@@ -0,0 +1,40 @@
+package org.tessellation.cutoff
+
+import org.tessellation.schema.SnapshotOrdinal
+
+import eu.timepit.refined.types.numeric.NonNegLong
+
+object LogarithmicOrdinalCutoff {
+  def make: OrdinalCutoff = new OrdinalCutoff {
+    def cutoff(cutoffOrdinal: SnapshotOrdinal, currentOrdinal: SnapshotOrdinal): List[SnapshotOrdinal] = {
+      val lastOrdinal = currentOrdinal.value.value
+
+      /* NOTE:
+       * To determine the necessary number of layers, Math.log10 should be used, although it necessitates a Double as its input type.
+       *   Math.log10(lastOrdinal) + 1
+       * Converting SnapshotOrdinal to Double may result in precision loss for substantial values.
+       * Given that SnapshotOrdinal is a non-negative Long, assesing the number of digits is sufficient to ascertain the number of layers.
+       */
+      val layers = lastOrdinal.toString.length
+
+      val (ordinalsToKeep, _) = (0 until layers).toList
+        .foldLeft((List.empty[Long], lastOrdinal)) {
+          case ((acc, lastOrdinal), layer) =>
+            val divident = Math.pow(10, layer.toDouble).toInt
+
+            val startRange = Math.max(0, lastOrdinal - Math.pow(10, layer.toDouble + 1).toInt + 1)
+
+            val endRange = lastOrdinal / divident.toInt
+            val ordinalsForLayer = (startRange / divident.toInt to endRange)
+              .map(_ * divident.toInt)
+              .toList
+
+            (acc ::: ordinalsForLayer, startRange)
```

Comment:
```suggestion
            (acc.concat(ordinalsForLayer), startRange)
```
This operation is much faster with a set, and we can throw out the intermediate `.toList` conversion

Reply:
:+1: